libs/langchain-mistralai/src/chat_models.ts

```diff
@@ -26,13 +31,51 @@ import {
 } from "@langchain/core/outputs";
 import { getEnvironmentVariable } from "@langchain/core/utils/env";
 import { NewTokenIndices } from "@langchain/core/callbacks/base";
+import { StructuredToolInterface } from "@langchain/core/tools";
+import { convertToOpenAITool } from "@langchain/core/utils/function_calling";
 
 interface TokenUsage {
   completionTokens?: number;
   promptTokens?: number;
   totalTokens?: number;
 }
 
+type MistralAIInputMessage = {
+  role: string;
+  name?: string;
+  content: string | string[];
+  tool_calls?: MistralAIToolCalls[];
+};
+
+type MistralAIToolInput = { type: string; function: MistralAIFunction };
+
+type MistralAIChatCompletionOptions = {
+  model: string;
+  messages: Array<{
+    role: string;
+    name?: string;
+    content: string | string[];
+    tool_calls?: MistralAIToolCalls[];
+  }>;
+  tools?: Array<MistralAIToolInput>;
+  temperature?: number;
+  maxTokens?: number;
+  topP?: number;
+  randomSeed?: number;
+  safeMode?: boolean;
+  safePrompt?: boolean;
+  toolChoice?: MistralAIToolChoice;
+  responseFormat?: ResponseFormat;
+};
+
+interface MistralAICallOptions extends BaseLanguageModelCallOptions {
+  responseFormat?: {
```

Comment:
Is this new? How does it interact with `tool_choice`?

Reply:
whoops I think I need to make this `response_format`. Same as in openai (json mode)